### DISCUSSION TITLE
Availability of Kubernetes patch releases?

### DISCUSSION THREAD

**Discussion by kastl-ars** [siderolabs/talos#9108] [Q&A] [answered]
Hi all,

this maybe a silly question, but the [support matrix](https://www.talos.dev/v1.7/introduction/support-matrix/) only mentions the Kubernetes minor versions (1.30, 1.29, 1.28, 1.27, 1.26, 1.25) that are supported.

Is there a place where I can lookup which **patch** releases are supported? My cluster is running 1.30.1, where do I look if there is a newer version (1.30.2, 1.30.3) available?

Or is this just using the upstream Kubernetes, so as soon as Kubernetes releases a new patch version, I can use it in Talos?

Kind Regards
Johannes

**Comment by smira** [comment, marked as answer]
All patch releases are supported within the major/minor supported release.

Talos Linux runs vanilla upstream Kubernetes images for the control plane components, so you can always assume those are available as soon as the release is available.

But the `kubelet` image is [built by us](https://github.com/siderolabs/kubelet) (Sidero Labs), so there's a slight delay (usually within 24 hours) before the image is available.

**Reply by kastl-ars** [reply]
Thanks for the quick reply! So once the kubelet image is available I could upgrade to a new patch release?
(Leaving out testing and stuff like that, just in principle).

Not sure if GitHub has a RSS feed for the package builds, so if there is a new release in the kubelet repository soon[tm] after there should be an image and I could update?

**Reply by smira** [reply]
yes, exactly. the upgrade would fail on image pre-pull if the image is not available (before making any changes).

**Reply by kastl-ars** [reply]
Ah, that is nice. I found the `pre-pull` in the help output and was delighted, as it seemed a very nice feature!

Thanks for your quick help, @smira
